scencrypt-hook

```diff
@@ -41,6 +41,34 @@ run_hook() {
     modprobe -a -q dm-crypt >/dev/null 2>&1
     [ "${quiet}" = "y" ] && CSQUIET=">/dev/null"
 
+    # Get plain key file if specified, example: cryptkey=/dev/sda:2048:4096, cryptkey=rootfs:/keyfile.bin
+    ckeyfile=
+    if [ -n "$cryptkey" ]; then
+        IFS=: read ckdev ckarg1 ckarg2 <<EOF
+$cryptkey
+EOF
```

Comment:
Any idea if we can switch to the format of:

```shellscript
IFS=: read ckdev ckarg1 ckarg2 <<< "$cryptkey"
```

for this? I know bash supports it, not totally sure about busybox ash.

Reply:
IIRC busybox ash is missing that feature, corrct me if I'm wrong. I copied that functionality from the default encrypt hook.